lib/Extended_ast.mli

```diff
@@ -24,7 +24,7 @@ type 'a t =
   | Expression : expression t
 
 module Parse : sig
-  val ast : 'a t -> Lexing.lexbuf -> 'a
+  val ast : 'a t -> conf:Conf.t -> Lexing.lexbuf -> 'a
```

Comment:
Passing the `Conf.t` record entirely for one option feels a bit out of scope.
I think it would be clearer with a smaller "conf" record defined in this module.